lib/rspec/support/output_formatter.rb

```diff
@@ -0,0 +1,46 @@
+module RSpec
+  module Support
+    # Provide additional output details beyond what `inspect` provides when
+    # printing Time, DateTime, or BigDecimal
+    module OutputFormatter
+      # @api private
+      def self.format(object)
+        if Time === object
+          format_time(object)
+        elsif defined?(DateTime) && DateTime === object
+          format_date_time(object)
+        elsif defined?(BigDecimal) && BigDecimal === object
+          "#{object.to_s 'F'} (#{object.inspect})"
+        else
+          object.inspect
+        end
+      end
```

Comment:
I think this method should also handle the case of calling `description` on an rspec matcher that implements `description`.  Currently we have this method for that:

https://github.com/rspec/rspec-support/blob/749e03b11ea78eaa643488328d1dc31f167af96f/lib/rspec/support/matcher_definition.rb#L34-L39

rspec-expectations has something similar:

https://github.com/rspec/rspec-expectations/blob/08c11431771b7755d2a402746b0fe4f8aec8d4ff/lib/rspec/matchers/composable.rb#L82-L85

Those both have a common purpose with this method in that they are trying to format an object for display in failure output.  Ideally, that would all be centralized in this one place so that all parts of RSpec benefit from common object formatting improvements.